lib/memo_wise.rb

```diff
@@ -423,6 +423,28 @@ def #{method_name}#{args_str}
           target.singleton_class.send(:private, method_name)
         end
       end
+
+      # Override [Module#instance_method](https://ruby-doc.org/core-3.0.0/Module.html#method-i-instance_method)
+      # to proxy the original `UnboundMethod#parameters` results. This proxying
+      # helps code that uses Ruby reflection over method parameters
+      def target.instance_method(symbol)
+        original_memo_wised_name = :"_memo_wise_original_#{symbol}"
+        curr_method = super
+
+        orig_method =
+          if symbol == :initialize
+            curr_method.super_method
```

Comment:
The fact that this evaluates to `super.super_method` is kind of confusing to me. Would you mind adding some comments here for what this is doing and why we need to special-case it?

Reply:
I'll add explicit documentation of the three paths:

1.  `initialize` - call `super` which is the original implementation of `Module#instance_method`, get back the `UnboundMethod` which represents our `MemoWise` version of `initialize` -- then ask **it** for `#super_method`, because it know the original method that it is overriding
 
2. Any memo-wised method - we need to look up the original method by it's renamed symbol, and call `super` which is the original implementation of `instance_method`, using that name

3. Anything else - we have `orig_method = nil`

Reply:
Modified per @jemmaissroff above and documented per this discussion, please re-review?